cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaValidator/index.tsx

```diff
@@ -66,13 +67,13 @@ class SchemaValidatorProvider extends React.Component<ISchemaValidatorProviderPr
   };
 
   public componentWillReceiveProps(nextProps) {
-    const { errors } = nextProps;
-    if (errors === null) {
+    const { errors, reset } = nextProps;
+    if (reset) {
```

Comment:
might not be a concern and also not sure what should the alternative be either, but is there a potential for a race condition here? where the props for the errors map gets updated first, and then the reset props set to false in a different digest cycle? that could potentially erase the errorMap

Reply:
Hhmm from the angular side we reset the validating flag and at the same time populate the error map.. React will batch state updates which will then trigger a prop update. So SchemaValidator componentWillReceiveProps will be called only after the setState in schemaeditor is done which will be batched even if the properties update async. This is what I noticed while testing as well. Let me know if I am missing anything.